shared/AppInsightsCore/Tests/Unit/src/W3cTraceParentTests.ts

```diff
@@ -206,5 +209,71 @@ export class W3cTraceParentTests extends AITestClass {
                 } as ITraceParent));
             }
         });
+
+        this.testCase({
+            name: "org generate guid timings",
+            useFakeTimers: false,
+            test: () => {
+                let start = dateNow();
+                for (let lp = 0; lp < 10000; lp++) {
+                    let newId = generateW3CId();
+                }
+
+                let orgTotal = dateNow() - start;
+
+                Assert.ok(orgTotal < 200, "Total time for org generateW3cID = " + JSON.stringify({
+                    orgTotal
+                }));
+            }
+        });
+
+        this.testCase({
+            name: "Validate generation",
+            useFakeTimers: false,
+            test: () => {
+                let key1 = generateW3CId();
+                let key2 = generateW3CId();
+
+                Assert.notEqual(key1, key2, "Make sure they are not the same");
+                Assert.equal(32, key1.length, "Key1 length: " + key1);
+                Assert.equal(32, key2.length, "Key2 length: " + key2);
+
+                Assert.equal("4", key1[12]);
+                Assert.equal("4", key2[12]);
+            }
+        });
+
+        this.testCase({
+            name: "generate newGuid timings",
+            useFakeTimers: false,
+            test: () => {
+                let start = dateNow();
+                for (let lp = 0; lp < 10000; lp++) {
+                    let newId = newGuid();
+                }
+
+                let guidTotal = dateNow() - start;
+
+                Assert.ok(guidTotal < 200, "Total time for org newGuid = " + JSON.stringify({
```

Comment:
While locally this occurs in around 8ms the build servers are virtualized and slower